src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -209,10 +273,23 @@ public Stream<String> getBlobsLinesStream(
   @Override
   public void validateUserCanRead(
       List<String> sourcePaths, String cloudEncapsulationId, AuthenticatedUserRequest user) {
-    // This checked is not needed for Azure because we use signed URLS that by default check
-    // permissions
-    // Keeping this method because we do need to do this check for GCP in code that is shared
-    // between gcp and azure [See CloudFileReader]
+    // This check is not needed for Azure source because we use signed URLS that by default check
+    // permissions but this is needed if ingesting from a GCS hosted file
+    List<String> gsPaths = sourcePaths.stream().filter(p -> p.startsWith("gs://")).toList();
+
+    // Extract the project ids from the userProject query parameter.  There can be 0 or 1 values
+    List<String> cloudEncapsulationIds =
+        gsPaths.stream()
+            .map(p -> UriUtils.getValueFromQueryParameter(p, USER_PROJECT_QUERY_PARAM))
+            .filter(Objects::nonNull)
+            .distinct()
+            .toList();
```

Comment:
You could use `count()` here instead of `toList()`

Reply:
eh, I need the list below